staging/src/k8s.io/csi-api/pkg/apis/csi/v1alpha1/README.md

```diff
@@ -0,0 +1,33 @@
+# CSINodeInfo and CSIDriverInfo Usage and Lifecycle
+CSINodeInfo is an API object representing CSI Information at the Node level.
+CSIDriverInfo is a subtype of CSINodeInfo that represents information about a
+driver at the node level. The CSIDriverInfoSpec represents the specification of
+the driver and is generally not changed, whereas the status represents the
+current status of the driver and can be updated and reacted to by various
+components.
+
+## Who creates it and when
+CSINodeInfo is created by Kubelet when the first CSI Driver is installed to the
+cluster and it is registered through the Kubelet device registration mechanism.
+
+## Who updates CSIDriverInfo Spec and when
+The Spec for CSIDriverInfo is created upon installation of the CSI Driver to the
+cluster and it is registered through the Kubelet device registration mechanism.
+The spec is populated with information about the driver through the
+nodeinfomanager and will remain unchanged from then on.
+
+## Who updates Status and when
+The Status for CSIDriverInfo is created upon installation of the CSI Driver to
+the cluster and it is registered through the Kubelet device registration
+mechanism. The Status contains information about installation and migration
+status of the driver. When the driver is installed/uninstalled through the
+Kubelet device registration mechanism the Installed flag is flipped from
+true/false respectively. The migration status will also be updated when the
+flags for migration are set to true/false on the Kubelet for that Driver on that
+node.
+
+## Consumers of Status and Spec
+Currently the only consumer of CSINodeInfo/CSIDriverInfo is the
+csi-external-provisioner. In the future, the Attach Detach Controller will need
+to read this object to determine migration status on a per driver per node
+basis.
```

Comment:
Discuss the possibility of a race?

Reply:
Added some to this section, and a new section about dealing with versioning